packages/adblocker/src/engine/engine.ts

```diff
@@ -1024,6 +1024,169 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
       };
     }
 
+    const { matches, allowGenericHides } = this.matchCosmeticFilters({
+      url,
+      hostname,
+      domain,
+      classes,
+      hrefs,
+      ids,
+      getRulesFromDOM,
+      getRulesFromHostname,
+      getInjectionRules,
+      getExtendedRules,
+      getPureHasRules: injectPureHasSafely,
+      callerContext,
+    });
+
+    const filters = [];
+
+    for (const { filter, exception } of matches) {
+      if (exception === undefined) {
+        filters.push(filter);
+      }
+    }
+
+    const { extended, scripts, styles } = this.injectCosmeticFilters(filters, {
+      url,
+      injectScriptlets: getInjectionRules,
+      injectExtended: getExtendedRules,
+      injectPureHasSafely,
+      allowGenericHides,
+      getBaseRules,
+      hidingStyle,
+    });
+
+    return {
+      active: true,
+      extended,
+      scripts,
+      styles,
+    };
+  }
+
+  public injectCosmeticFilters(
+    filters: CosmeticFilter[],
+    {
+      url,
+
+      injectStyles = true,
+      injectScriptlets,
+      injectExtended,
+      injectPureHasSafely,
+
+      allowGenericHides = true,
+      getBaseRules,
+      hidingStyle,
+    }: {
+      url: string;
+
+      injectStyles?: boolean;
+      injectScriptlets: boolean;
+      injectExtended: boolean;
+      injectPureHasSafely: boolean;
+
+      allowGenericHides?: boolean;
+      hidingStyle?: string | undefined;
+      getBaseRules?: boolean;
+    },
+  ): {
+    scripts: string[];
+    extended: IMessageFromBackground['extended'];
+    styles: string;
+  } {
+    const scripts = [];
+    const styleFilters = [];
+    const extendedFilters = [];
+    const pureHasFilters = [];
+
+    for (const filter of filters) {
+      if (injectScriptlets && filter.isScriptInject()) {
+        const script = filter.getScript(this.resources.getScriptlet.bind(this.resources));
+        if (script !== undefined) {
+          scripts.push(script);
+        }
+      } else if (filter.isExtended()) {
+        if (injectExtended === true && this.config.loadExtendedSelectors) {
+          extendedFilters.push(filter);
+        }
+        if (injectPureHasSafely && filter.isPureHasSelector()) {
+          pureHasFilters.push(filter);
+        }
+      } else if (injectStyles === true) {
+        styleFilters.push(filter);
+      }
+    }
+
+    const stylesheets = this.cosmetics.getStylesheetsFromFilters(
+      {
+        filters: styleFilters,
+        extendedFilters,
+      },
+      { getBaseRules, allowGenericHides, hidingStyle },
+    );
+
+    let styles = stylesheets.stylesheet;
+    for (const filter of pureHasFilters) {
+      styles += `\n\n${createStylesheet([filter.getSelector()], hidingStyle)}`;
+    }
+
+    for (const script of scripts) {
+      this.emit('script-injected', script, url);
+    }
+
+    if (styles.length !== 0) {
+      this.emit('style-injected', styles, url);
+    }
```

Comment:
I think these were meant to be same as `filter-matched` in previous. Wouldn't it be more clear to move these to `matchCosmeticFilters`?

Reply:
those logs require "compiled" stylesheets and scripts, so cannot be emitted on match as those representations does not exist yet